docs/source/faq.rst

```diff
@@ -159,27 +159,24 @@ Second, if your objective function behaves in a non-deterministic way (i.e., it
 To deal with this problem, please set an option (e.g., random seed) to make the behavior deterministic if your optimization target (e.g., an ML library) provides it.
 
 
-How does Optuna handle NaNs and exceptions reported by the objective function?
-------------------------------------------------------------------------------
+How are exceptions from trials handled?
+---------------------------------------
+
+Trials that raise, without catching, exceptions will be treated as failures, i.e. with the :obj:`~optuna.structs.TrialState.FAIL` status.
 
-Optuna treats such trials as failures (i.e., :obj:`~optuna.structs.TrialState.FAIL`) and continues the study.
-The Optuna's system process will not be crashed by any objective values or exceptions raised in objective functions.
+By default, all exceptions raised in objective functions are propagated to user code.
```

Comment:
I think that is not true in the case of [`TrialPruned`](https://optuna.readthedocs.io/en/stable/reference/exceptions.html#optuna.exceptions.TrialPruned). It is a kind of exceptions, but it does not stop `Study.optimize`.

In addition, `user code` is a bit confusing for me because the user code is usually nested, i.e., the outer user code invokes `Study.optimize`, and the inner user code describes the objective function. IMHO, the sentence `studies are aborted when exceptions are raised.` is clearer than this sentence.

Reply:
Do you mean to omit this sentence?

Reply:
How about something along the following?

```rst
Trials that raise exceptions without catching them will be treated as failures, i.e. with the :obj:`~optuna.structs.TrialState.FAIL` status.

By default, all exceptions except :class:`~optuna.exceptions.TrialPruned` raised in objective functions are propagated to the caller of :func:`~optuna.study.Study.optimize`.
In other words, studies are aborted when such exceptions are raised.
It might be desirable to continue a study with the remaining trials.
To do so, you can specify in :func:`~optuna.study.Study.optimize` which exceptions types to catch using the ``catch`` argument.
Exceptions of these types are caught inside the study and will not propagate further.
```

Reply:
I think this description is accurate, both in terms of exceptions and the user code.
Thank you for your suggestion.

> How about something along the following?
> 
> ```
> Trials that raise exceptions without catching them will be treated as failures, i.e. with the :obj:`~optuna.structs.TrialState.FAIL` status.
> 
> By default, all exceptions except :class:`~optuna.exceptions.TrialPruned` raised in objective functions are propagated to the caller of :func:`~optuna.study.Study.optimize`.
> In other words, studies are aborted when such exceptions are raised.
> It might be desirable to continue a study with the remaining trials.
> To do so, you can specify in :func:`~optuna.study.Study.optimize` which exceptions types to catch using the ``catch`` argument.
> Exceptions of these types are caught inside the study and will not propagate further.
> ```

Reply:
Applied!